src/content/interop/c-interop.md

```diff
@@ -79,269 +86,296 @@ Hello World
-7. Get a reference to the C function, and put it into a variable.
+
+   ```dart
+   final dylib = ffi.DynamicLibrary.open(libraryPath);
+   ```
+
+7. Get a reference to the C function,
+   and put it into a variable.
+   This code uses the `typedefs` from steps 2 and 3,
+   along with the dynamic library variable from step 4.
+   ```dart
+   final HelloWorld hello = dylib
+       .lookup<ffi.NativeFunction<hello_world_func>>('hello_world')
+       .asFunction();
+   ```
+
 8. Call the C function.
 
-Here's the code for each step.
-
- 1. Import `dart:ffi`.
- 
-    ```dart
-    import 'dart:ffi' as ffi;
-    ```
- 
- 2. Import the path library that you'll use to store the path of dynamic library.
- 
-    ```dart
-    import 'dart:io' show Platform, Directory;
-    import 'package:path/path.dart' as path;
-    ```
- 
- 3. Create a typedef with the FFI type signature of the C function. <br>
-    See [Interfacing with native types](#interfacing-with-native-types)
-    for commonly used types defined by `dart:ffi` library.
- 
-    ```dart
-    typedef hello_world_func = ffi.Void Function();
-    ```
- 
- 4. Create a typedef for the variable that you'll use
-    when calling the C function.
- 
-    ```dart
-    typedef HelloWorld = void Function();
-    ```
- 
- 5. Create a variable to store the path of the dynamic library.
- 
-    ```dart
-    var libraryPath = path.join(Directory.current.path, 'hello_library',
-        'libhello.so');
-    if (Platform.isMacOS) { 
-      libraryPath = path.join(Directory.current.path, 'hello_library', 
-          'libhello.dylib');
-    } else if (Platform.isWindows) { 
-      libraryPath = path.join(Directory.current.path, 'hello_library', 
-          'Debug', 'hello.dll');
-    } 
-    ```
- 
- 6. Open the dynamic library that contains the C function.
- 
-    ```dart
-    final dylib = ffi.DynamicLibrary.open(libraryPath);
-    ```
- 
- 7. Get a reference to the C function, 
-    and put it into a variable.
-    This code uses the typedefs defined in steps 2 and 3, 
-    along with the dynamic library variable from step 4.
-    
-    ```dart
-    final HelloWorld hello = dylib
-        .lookup<ffi.NativeFunction<hello_world_func>>('hello_world')
-        .asFunction();
-    ```
- 
- 8. Call the C function.
- 
-    ```dart
-    hello();
-    ```
-
-Once you understand the hello_world example, 
-you should be ready to look at the
-[other `dart:ffi` examples](#examples).
-
-
-## Bundling and loading C libraries
-
-How you bundle (or _package_ or _distribute_)
-a C library with your package or app
-and then load that library
-depends on your platform and the type of library.
-For details, see the following:
-
-* Flutter `dart:ffi` pages: [Android][android], [iOS][ios], and [macOS][macos] 
+   ```dart
+   hello();
+   ```
+
+Once you understand the `hello_world` example,
+consult the [other `dart:ffi` examples](#download-example-files).
+
+## Bundle and load C libraries
```

Comment:
I suppose we mostly want to reference what is non-experimental here.

The work on native assets (which is currently an experiment in Dart and Flutter), will make bundling / distributing native code platform agnostic.

* https://github.com/dart-lang/sdk/issues/50565
* https://github.com/flutter/flutter/issues/129757

Edit: Ah we already have a section for native assets, maybe link it here.

Reply:
Already done on line [264](https://github.com/dart-lang/site-www/pull/5881/files#diff-32de1796331bba2e5ab22ab36136524d890f3de00d3b6d1a16ceb32c55f92407R264)